### PR TITLE
chore(core): track copy block to link action

### PR DESCRIPTION
**Pull request by fundon**
part of [BS-1254](https://linear.app/affine-design/issue/BS-1254/增加-block-reference-的埋点)

### CODE REVIEW

**Comment by fundon** [comment]
* **#8056** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8056?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2  dependent PRs ([#8060](https://github.com/toeverything/AFFiNE/pull/8060) <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8060?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#8087](https://github.com/toeverything/AFFiNE/pull/8087) <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8087?utm_source=stack-comment-icon" target="_blank">Graphite</a>) 👈
* **#7946** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7946?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2 other dependent PRs ([#8001](https://github.com/toeverything/AFFiNE/pull/8001) <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8001?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#8002](https://github.com/toeverything/AFFiNE/pull/8002) <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8002?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#7933** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7933?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7886** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7886?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @fundon and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/core/src/components/blocksuite/block-suite-editor/specs/custom/database-block.ts [hunk 1]
```diff
@@ -71,6 +72,14 @@ function createCopyLinkToBlockMenuItem(
       const str = generateUrl(options);
       if (!str) return;
 
+      mixpanel.track('CopyBlockLink', {
```
Comment (forehalo) [inline packages/frontend/core/src/components/blocksuite/block-suite-editor/specs/custom/database-block.ts]: use `track` from `@affine/core/mixpanel`

File: packages/frontend/core/src/components/blocksuite/block-suite-editor/specs/custom/widgets/toolbar.ts [hunk 2]
```diff
@@ -74,27 +79,50 @@ function createCopyLinkToBlockMenuItem(
...
       const pageId = editor.doc.id;
       const workspaceId = editor.doc.workspace.id;
       const options: UseSharingUrl = { workspaceId, pageId, shareMode: mode };
+      let type = '';
 
       if (mode === 'page') {
         // maybe multiple blocks
         const blockIds = ctx.selectedBlockModels.map(model => model.id);
         options.blockIds = blockIds;
+        type = ctx.selectedBlockModels[0].flavour.split(':')[1];
```
Comment (forehalo) [inline packages/frontend/core/src/components/blocksuite/block-suite-editor/specs/custom/widgets/toolbar.ts]: it means to remove the 'affine:' prefix right? I think it's ok to keep it

Comment (fundon) [reply packages/frontend/core/src/components/blocksuite/block-suite-editor/specs/custom/widgets/toolbar.ts]: yes, block/element's type is required

**Review by pengx17 — APPROVED** [review]

**Review by EYHN — APPROVED** [review]

**Pull request merged**